python/taichi/lang/matrix.py

```diff
@@ -801,14 +786,13 @@ def cross(self, b):
                 "Cross product is only supported between pairs of 2D/3D vectors"
             )
 
-    @taichi_scope
     def outer_product(self, b):
         assert self.m == 1
         assert b.m == 1
         c = Matrix(self.n, b.n)
         for i in range(self.n):
             for j in range(b.n):
-                c(i, j).assign(self(i) * b(j))
+                c.set_entry(i, j, self(i) * b(j))
```

Comment:
I think this changes the behavior in Taichi-scope. We probably want `assign` there.